runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SpecMonitoringInfoValidator.java

```diff
@@ -68,6 +68,7 @@ public SpecMonitoringInfoValidator() {
               monitoringInfo.getUrn(), spec.getTypeUrn(), monitoringInfo.getType()));
     }
 
+    // TODO(ajamato): Tighten this restriction to use set equality, to catch unused
```

Comment:
Just noting: I recently had a local change to keep this a bit relaxed, because I was testing with user metrics that were getting a `PTRANSFORM` label. I'd weakened the condition below to: 

```java
!requiredLabels.ieEmpty && !monitoringInfo.getLabelsMap().keySet().equals(requiredLabels)
```

where previously it required strict equality in all cases. 

Seemingly this was relaxed to the `containsAll` below before I got anywhere with my change, and it's possible I was in some other invalid state to have had user metrics with a `PTRANSFORM` label that were failing the strict test (where `requiredLabels` was empty).

Just wanted to leave a breadcrumb here about that since we'll probably come back to it soon..